File: src/main/java/net/alephdev/calendar/dto/functional/TopRiskDto.java
package net.alephdev.calendar.dto.functional;

import io.swagger.v3.oas.annotations.media.Schema;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
@Schema(description = "Агрегация потерь по риску для топ-10 отчета")
public class TopRiskDto {
  @Schema(description = "ID риска", example = "7")
  private final Integer riskId;

  @Schema(
      description = "Описание риска",
      example = "Падение производительности при высокой нагрузке")
  private final String description;

  @Schema(description = "Суммарные ожидаемые потери", example = "15234.50")
  private final BigDecimal totalEstimatedLoss;

  public TopRiskDto(TopRiskProjection projection) {
    this(projection.getRiskId(), projection.getDescription(), projection.getTotalEstimatedLoss());
  }
}

File: angular/src/app/models/dto/top-risk-dto.ts
/**
 * DTO для представления агрегированной информации по риску в топ‑10.
 *
 * @property riskId - Идентификатор риска
 * @property description - Описание риска
 * @property totalEstimatedLoss - Суммарные потенциальные потери
 */
export interface TopRiskDto {
  riskId: number;
  description: string;
  totalEstimatedLoss: number;
}
